site/zenodo_rdm/support/support.py

```diff
@@ -0,0 +1,120 @@
+import smtplib
+from collections import OrderedDict
+
+from flask import current_app, redirect, render_template, request, url_for
+from flask.views import MethodView
+from flask_login import current_user
+from invenio_accounts.sessions import _extract_info_from_useragent
+from marshmallow import ValidationError
+from werkzeug.utils import cached_property
+
+from .email import SupportEmailService
+from .errors import ConfirmationEmailNotSent, SupportEmailNotSent
+from .schema import SupportFormSchema
+
+
+class ZenodoSupport(MethodView):
+    """Zenodo support view."""
+
+    def __init__(self):
+        self.template = "zenodo_rdm/support.html"
+        self.email_service = SupportEmailService()
+        self.support_form_schema = SupportFormSchema()
+
+    def get(self):
+        """Renders the support template."""
+        user_agent = _extract_info_from_useragent(request.headers.get("User-Agent"))
+        browser_client = user_agent.get("browser", "")
+        browser_version = user_agent.get("browser_version", "")
+        browser_string = browser_client + " " + browser_version
+        platform = user_agent.get("os", "")
+        system_info = {"browser": browser_string, "platform": platform}
+
+        return render_template(
+            self.template,
+            categories=self.categories,
+            system_info=system_info,
+        )
+
+    def post(self):
+        """Receives a form, validates its data and handles it."""
+        data = {**request.form.to_dict(), "files": request.files.getlist("files")}
+
+        try:
+            data = self.validate_form(data)
+        except (ValidationError) as e:
+            return str(e), 400
+
+        try:
+            self.handle_form(data)
+        except (Exception) as e:
+            return str(e), 500
+
+        return redirect(url_for("invenio_app_rdm.index"))
+
+    def handle_form(self, form_data):
+        """Form controller."""
```

Comment:
Currently the only handler is to send two emails. However it can grow into different behaviors (e.g. based on a category do something else)